src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the process information from the Orsay server and saves it in the processInfo VA
+        """
+        if parameter is None:
+            parameter = self.datamodel.HybridPlatform.ProcessInfo
+        if parameter is not self.datamodel.HybridPlatform.ProcessInfo:
+            raise ValueError("Incorrect parameter passed to _updateProcessInfo. Parameter should be "
+                             "datamodel.HybridPlatform.ProcessInfo. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        currentProcessInfo = str(parameter.Actual)
+        currentProcessInfo = currentProcessInfo.replace("N/A", "")
+        logging.debug("ProcessInfo update: " + currentProcessInfo)
+        self.processInfo._set_value(currentProcessInfo, force_write=True)
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        if self._device:
+            self.datamodel.HybridPlatform.ProcessInfo.Unsubscribe(self._updateProcessInfo)
+            if self._pneumaticSuspension:
+                self._pneumaticSuspension.terminate()
+                self._pneumaticSuspension = None
+            if self._pressure:
+                self._pressure.terminate()
+                self._pressure = None
+            if self._pumpingSystem:
+                self._pumpingSystem.terminate()
+                self._pumpingSystem = None
+            if self._ups:
+                self._ups.terminate()
+                self._ups = None
+            super(OrsayComponent, self).terminate()
+            self._stop_connection_monitor.set()  # stop trying to reconnect
+            self._device.HttpConnection.close()  # close the connection
+            self._device.MessageConnection.Connection.close()
+            self._device = None
+            self.datamodel = None
+
+
+class pneumaticSuspension(model.HwComponent):
+    """
+    This represents the Pneumatic Suspension from Orsay Physics
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Defines the following VA's and links them to the callbacks from the Orsay server:
+        • power (BooleanVA, value corresponds to _valve.Actual == VALVE_OPEN, set to True to open/start and False to
+        close/stop)
+        • pressure (FloatContinuous, range=(0, 110000), read-only, unit is "Pa", value is _gauge.Actual)
+        """
+
+        # we will fill the set of children with Components later in ._children
+        model.HwComponent.__init__(self, name, role, parent=parent, **kwargs)
+
+        self._valve = None
+        self._gauge = None
+
+        self.pressure = model.FloatContinuous(0.0, range=(0, 110000), readonly=True, unit="Pa")
```

Comment:
Same comment as for vacuum chamber range. Is expected that they have the same maximum and minimum? I assume that the pneumatic table would go above the normal atmospheric pressure, so anything below would be very odd, especially 0...